posts/2023-12-30_containers_webassembly_submission/containers_and_webassembly_submissions.qmd

```diff
@@ -0,0 +1,255 @@
+
+### The Journey with WebAssembly and Containers
+
+Our team at Appsilon teamed up with the dynamic Pilot 4 crew to explore WebAssembly technology and containers. George Stagg and Winston Chang also joined the working group to discuss the web-assembly portion of Pilot 4. This partnership brought together our engineering prowess to contribute to these tools, injecting fresh perspectives into the ongoing pilot project.
+
+Some of the outcomes of the collaboration:
+
+1.  We were able to set up a robust container environment for this pilot project. 
+
+2.  We aided the progress made on the use of both experimental technologies: containers and WebAssembly.
+
+3.  We developed a working prototype submission using [Podman](https://podman.io/) container technology.
+
+4.  We developed a working early-stage prototype for wrapping a small Shiny application using WebAssembly.
+
+#### WebAssembly
+
+![](images/webr.png){width="231"}
+
+WebAssembly allows languages like R to be executed at near-native speed directly within web browsers, providing users with the ability to run R code without having R installed locally. [**WebR**](https://docs.r-wasm.org/webr/v0.2.0/) is essentially the R programming language adapted to run in a web browser environment using WebAssembly. This project is under active development. 
+
+### The Pilot 4 Shiny App Up and Running on webR!
+
+![](images/rconsortiumxappsilon.png){fig-align="center"}
+
+The deployed example of the Shiny app running on webR [is available here](https://brilliant-elf-2da930.netlify.app/). Check out the video of the application running below.
+
+{{< video https://youtu.be/wfk_IJ82e9E >}}
+
+During this pilot, engineers at Appsilon developed a prototype of a Shiny application running on webR. The application reuses most of the code from the previous pilot apps with some tweaks and a couple of hacks/changes to get around non CRAN dependencies, specially for data loading, WebR compatibilities, and shimming some of the functionality from {teal} and other packages that are (for now) not available on CRAN.
+
+#### webR Shiny App
+
+During the **second iteration**, which was recently held, Pedro Silva shared the process of developing this Shiny app running on webR.
+
+#### The Process
+
+1.  **Leverage last 2 iterations of the application**
+
+    -   Reuse as much code as possible
+
+    -   Avoid touching the logic part
+
+2.  **Restrict the number of dependencies to packages on CRAN**
+
+    -   Replace/shim functionality that was lost from removing dependencies
+
+Here is the list of dependencies to packages on CRAN, those that worked are colored green, those that were removed are marked in orange. We ended up with just 3 problematic dependencies (bold).
+
+|                                           |                                           |
+|------------------------------------|------------------------------------|
```

Comment:
maybe this could be a small scrollable table?  Just takes up a lot of real estate